tlslite/tlsconnection.py

```diff
@@ -1307,7 +1307,17 @@ def _handshakeServerAsyncHelper(self, verifierDB,
                 if result in (0,1): yield result
                 else: break
             premasterSecret = result
-        
+
+        elif cipherSuite in CipherSuite.ecdhAnonSuites:
+            acceptedCurves = self._curveNamesToList(settings)
+            keyExchange = AECDHKeyExchange(cipherSuite, clientHello,
+                                           serverHello, acceptedCurves)
+            for result in self._serverAnonKeyExchange(serverHello, keyExchange,
+                                                      cipherSuite):
+                if result in (0,1): yield result
+                else: break
+            premasterSecret = result
```

Comment:
everything from `for result ...` is the same as in the ADH, the code can reuse it